packages/react/src/components/header/components/navigationLanguageSelector/NavigationLanguageSelector.tsx

```diff
@@ -2,67 +2,91 @@ import React, { PropsWithChildren, useEffect } from 'react';
 import classes from './NavigationLanguageSelector.module.scss';
-import { useHeaderContext } from '../../HeaderContext';
+import { useHeaderContext, useSetHeaderContext } from '../../HeaderContext';
 
 export type LanguageSelectorProps = PropsWithChildren<{
   /**
    * Array of languages as LanguageOption objects.
    */
-  languages: LanguageOption[];
+  languages?: LanguageOption[];
+  /**
+   * Mobile mode.
+   */
+  fullWidthForMobile?: boolean;
 }>;
 
-const LanguageButton = ({ language }) => {
+const LanguageButton = ({ value, label }: LanguageOption) => {
   const activeLanguage = useActiveLanguage();
   const setLanguage = useSetLanguage();
-  const className = classNames(classes.item, { [classes.activeItem]: activeLanguage === language.value });
-  const selectLanguage = withDefaultPrevented(() => setLanguage(language.value));
+  const className = classNames(classes.item, { [classes.activeItem]: activeLanguage === value });
+  const selectLanguage = withDefaultPrevented(() => setLanguage(value));
 
   return (
-    <button key={language.value} lang={language.value} onClick={selectLanguage} type="button" className={className}>
-      <span>{language.label}</span>
+    <button key={value} lang={value} onClick={selectLanguage} type="button" className={className}>
+      <span>{label}</span>
     </button>
   );
 };
 
-const renderLanguageNode = (language: LanguageOption) => <LanguageButton key={language.value} language={language} />;
+const renderLanguageNode = (language: LanguageOption) => {
+  return <LanguageButton key={language.value} value={language.value} label={language.label} />;
+};
 
-export const NavigationLanguageSelector = ({ children, languages }: LanguageSelectorProps) => {
+export const NavigationLanguageSelector = ({ children, languages, fullWidthForMobile }: LanguageSelectorProps) => {
   const setAvailableLanguages = useSetAvailableLanguages();
-  const { isNotLargeScreen } = useHeaderContext();
+  const { isNotLargeScreen, languageSelectorContent } = useHeaderContext();
+  const { setLanguageSelectorContent } = useSetHeaderContext();
+
+  const isInActionBar = !fullWidthForMobile;
+  const isRender = !languages;
+  if (!isRender) {
+    useEffect(() => {
+      setAvailableLanguages(languages);
+    }, [languages]);
+  }
 
-  useEffect(() => {
-    setAvailableLanguages(languages);
-  }, [languages]);
+  if (children) {
+    useEffect(() => {
+      setLanguageSelectorContent(getChildElementsEvenIfContainersInbetween(children));
+    }, [children]);
+  }
 
-  const languageNodes = languages.map(renderLanguageNode);
+  const languageNodes = (languages || useAvailableLanguages()).map(renderLanguageNode);
 
   return (
-    <div className={classes.languageSelector}>
-      {!isNotLargeScreen && languageNodes}
+    (isRender && isNotLargeScreen !== isInActionBar && (
+      <div className={classNames(classes.languageSelector, { [classes.fullWidthForMobile]: !isInActionBar })}>
```

Comment:
In mobile view, the language bar should have a background colour and the language menu icons should be on the right hand corner. [Quick link to designs](https://www.figma.com/file/Kxwg3R0zNRHj55nQqFu6VS/HDS-UI-Kit-3.0?node-id=1117%3A28174&mode=dev)

Reply:
If there are no languages and no NavigationLanguageSelector component declared, this component's styles are still in effect.
- on desktop the menu icon is showing
- on mobile the border-bottom is showing, creating a double thick bottom border for action bar